### ISSUE TITLE
Rational powers and coversion

### ISSUE THREAD

**Issue by pwl** [JuliaLang/julia#18114]
There is something wrong with `Rational` powers:
```
julia> BigFloat(2)^(BigFloat(1)/BigFloat(3))-BigFloat(2)^(1//3)
1.615949454734010502584703185359054123712092705937300412204179527347802183365638e-17
```
where there shouldn't be any difference (`1//3` is of infinite precision).  The issue comes from the [definition](https://github.com/JuliaLang/julia/blob/master/base/rational.jl#L378-L379) of `^` for rationals, where convert is applied to the result of a division of two integers, which is basically a `Float64`, the result is then converted back to `BigFloat`
```julia
^{T<:AbstractFloat}(x::T, y::Rational) = x^(convert(T, y.num / y.den))
^{T<:AbstractFloat}(x::Complex{T}, y::Rational) = x^(convert(T, y.num / y.den))
```

Redefining `^` like this
```julia
^{T<:AbstractFloat}(x::T, y::Rational) = x^(convert(T, y.num) / convert(T, y.den))
^{T<:AbstractFloat}(x::Complex{T}, y::Rational) = x^(convert(T, y.num) / convert(T, y.den))
```
fixes the problem:
```
julia> BigFloat(2)^(BigFloat(1)/BigFloat(3))-BigFloat(2)^(1//3)
0.000000000000000000000000000000000000000000000000000000000000000000000000000000
```

In summary, `convert` doesn't commute with division (nor with other similar operations) and `convert` should be applied as early as possible (directly to the integers in this case) to avoid precision loss.  There might be more cases like this that I am not aware of.

EDIT: This bug is new in 0.5 and it was introduced in #17530.

**Comment by simonbyrne** [comment]
Would you mind opening a pull request?
